command/server.go

```diff
@@ -1076,11 +1096,37 @@ func (c *ServerCommand) Run(args []string) int {
 		default:
 			devStorageType = "inmem"
 		}
-		config, err = server.DevConfig(devStorageType)
+
+		if c.flagDevTLS {
+			if c.flagDevTLSCertDir != "" {
+				_, err := os.Stat(c.flagDevTLSCertDir)
+				if err != nil && !os.IsNotExist(err) {
+					c.UI.Error(err.Error())
+					return 1
+				}
+
+				certDir = c.flagDevTLSCertDir
+			} else {
+				certDir, err = os.MkdirTemp("", "vault-tls")
+				if err != nil {
+					c.UI.Error(err.Error())
+					return 1
+				}
+			}
+			config, err = server.DevTLSConfig(devStorageType, certDir)
+
+			defer os.Remove(fmt.Sprintf("%s/%s", certDir, server.VaultCAFilename))
+			defer os.Remove(fmt.Sprintf("%s/%s", certDir, server.VaultCertFilename))
+			defer os.Remove(fmt.Sprintf("%s/%s", certDir, server.VaultKeyFilename))
```

Comment:
It may be a good idea to log the errors from the Removes (if any).

Reply:
Done!